components/server/src/config/configuration-service.ts

```diff
@@ -36,16 +36,22 @@ export class ConfigurationService {
         }
         // eagerly fetch for all files that the inferrer usually asks for.
         this.requestedPaths.forEach(path => !(path in cache) && readFile(path));
-        const config: WorkspaceConfig = await new ConfigInferrer().getConfig({
+        const configInferrer = new ConfigInferrer();
+        const config: WorkspaceConfig = await configInferrer.getConfig({
             config: {},
             read: readFile,
             exists: async (path: string) => !!(await readFile(path)),
         });
         if (!config.tasks) {
             return;
         }
-        const configString = `tasks:\n  - ${config.tasks.map(task => Object.entries(task).map(([phase, command]) => `${phase}: ${command}`).join('\n    ')).join('\n  - ')}`;
-        return configString;
+        const configString = configInferrer.toYaml(config);
+        return `# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/config-gitpod-file)
+# and commit this file to your remote git repository to share the goodness with others.
+
+${configString}
+`;
```

Comment:
No need to add a second empty line at the bottom of the file (can also be fixed in a follow-up):

```suggestion
${configString}`;
```